README update for resource naming (#2600)
Azure/ARO-HCP: test/e2e/README.md

For more information about ARO HCP environments, see the [ARO HCP Environments documentation](https://github.com/Azure/ARO-HCP/blob/main/docs/environments.md).

## Writting and running new E2E Test cases

### Resource Naming

> **Important:** These tests are running in parallel so it is **VITAL** that we avoid naming collisions with other tests that may be running in CI at the same time. This may break CI runs until the duplicate resources are removed! 
- The customer resource group name must be unique across the subscription. Using NewResourceGroup() from the framework will provide a unique resource group name as well as handle the cleanup.  

> - The combination of the cluster name and the managed resource group name must be unique across the subscription. If you use the provided bicep templates they will handle this by appending -rg to the customer resource group name.

Reviewer: ```suggestion
- The managed resource group name must be unique across the subscription. If you use the provided bicep templates they will handle this by appending `-managed` suffix to the customer resource group name to create a unique managed resource group name.
```

Reviewer: I'm suggesting a fix to better describe how Ben fixed the defaults in https://github.com/Azure/ARO-HCP/pull/2595/files
